### PR TITLE
feat: add a block inflater that supports recycling

### PR DESCRIPTION
**Pull request by gonfunko**
This PR adds a block flyout inflater that supports recycling, in order to improve performance when refreshing the toolbox.

### CODE REVIEW

File: src/recyclable_block_flyout_inflater.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from "blockly/core";
+
+/**
+ * A block inflater that caches and reuses blocks to improve performance.
+ */
+export class RecyclableBlockFlyoutInflater extends Blockly.BlockFlyoutInflater {
+  recyclingEnabled = true;
```
Comment (rachel-fenichel) [inline src/recyclable_block_flyout_inflater.js:17]: tsdoc on these two.

File: src/recyclable_block_flyout_inflater.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+   * @param {boolean} enabled True if recycling should be enabled.
+   */
+  setRecyclingEnabled(enabled) {
+    this.recyclingEnabled = enabled;
+  }
+
+  /**
+   * Creates a new block from the given block definition.
+   *
+   * @param {Object} blockDefinition The definition to create a block from.
```
Comment (rachel-fenichel) [inline src/recyclable_block_flyout_inflater.js]: You're mixing js-style and ts-style docs in this file (some have `{type}` and some don't. Stick with one.

File: src/recyclable_block_flyout_inflater.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  emptyRecycledBlocks() {
+    this.recycledBlocks
+      .values()
+      .forEach((block) => block.dispose(false, false));
+    this.recycledBlocks.clear();
+  }
+}
+
+/**
+ * Registers the recyclable block flyout inflater.
```
Comment (rachel-fenichel) [inline src/recyclable_block_flyout_inflater.js]: Add "replacing the standard block flyout inflator".

**Review by rachel-fenichel — CHANGES_REQUESTED** [review]

File: src/recyclable_block_flyout_inflater.js (hunk 3 above)
Comment (gonfunko) [reply src/recyclable_block_flyout_inflater.js]: Done

File: src/recyclable_block_flyout_inflater.js (hunk 2 above)
Comment (gonfunko) [reply src/recyclable_block_flyout_inflater.js]: Added the types.

File: src/recyclable_block_flyout_inflater.js (hunk 1 above)
Comment (gonfunko) [reply src/recyclable_block_flyout_inflater.js:17]: Done.

**Review by rachel-fenichel — APPROVED** [review]

**Pull request merged**